src/TimeManagementService.Validation/EditWorkTimeRequestValidator.cs

```diff
@@ -1,71 +1,161 @@
+                .WithMessage("You don't have to change the same field of WorkTime multiple times.")
+                .Must(x => x.Any())
+                .WithMessage("You don't have changes.")
+                .ForEach(x => x
+                .Must(x => Paths.Contains(x.path))
+                .WithMessage($"Document contains invalid path. Only such paths are allowed: {Paths.Aggregate((x, y) => x + ", " + y)}")
+                )
+                .DependentRules(() =>
                 {
-                    var oldWorkTimes = repository.GetUserWorkTimes(
-                        wt.WorkerUserId,
-                        new WorkTimeFilter
+                    When(wt => GetOperationByPath(wt.Patch, StartTimePath) != null, () =>
+                    {
+                        RuleFor(x => x.Patch.Operations)
+                        .UniqueOperationWithAllowedOp(StartTimePath, "add", "replace");
+
+                        RuleFor(wt => (DateTime)GetOperationByPath(wt.Patch, StartTimePath).value)
+                        .NotEqual(new DateTime())
+                        .WithMessage($"{StartTimePath} must not be empty.");
+                    });
+
+                    When(wt => GetOperationByPath(wt.Patch, EndTimePath) != null, () =>
+                    {
+                        RuleFor(x => x.Patch.Operations)
+                        .UniqueOperationWithAllowedOp(EndTimePath, "add", "replace");
+
+                        RuleFor(wt => (DateTime)GetOperationByPath(wt.Patch, EndTimePath).value)
+                        .NotEqual(new DateTime())
+                        .WithMessage($"{EndTimePath} must not be empty.");
+                    });
+
+                    When(wt => GetOperationByPath(wt.Patch, TitlePath) != null, () =>
+                    {
+                        RuleFor(x => x.Patch.Operations)
+                        .UniqueOperationWithAllowedOp(TitlePath, "add", "replace");
+
+                        RuleFor(wt => (string)GetOperationByPath(wt.Patch, TitlePath).value)
+                        .MaximumLength(32)
+                        .WithMessage($"{TitlePath} is too long.")
+                        .MinimumLength(2)
+                        .WithMessage($"{TitlePath} is too short.");
+                    });
+
+                    // Oh. Check overlap with new User? Admin can it?
+                    //When(wt => GetOperationByPath(wt.Patch, WorkerUserIdPath) != null, () =>
+                    //{
+                    //    RuleFor(x => x.Patch.Operations)
+                    //    .UniqueOperationWithAllowedOp(WorkerUserIdPath, "add", "replace");
+
+                    //    //check exist?
+
+                    //    RuleFor(wt => (Guid)GetOperationByPath(wt.Patch, WorkerUserIdPath).value)
+                    //    .NotEmpty()
+                    //    .WithMessage($"{WorkerUserIdPath} must not be empty.");
+                    //});
```

Comment:
comments?

Reply:
I am waiting for Spartak's answer to the question at the beginning of the comment